### PR TITLE
CondDBESSources: reworked refresh for updated lumi-based conditions

### PR DESCRIPTION
**Pull request by ggovi**
#### PR description:

The CondDBESSources module allows the consumption of lumi-based conditions, support the refresh in case of tags updated "live". The refresh logic, previously based on an "artificial" upper limit of the IOV from the Condition DB, has been changed and it is now entirely based on Framework features.   The refreshTime parameter is still required and set the maximum number of lumisections of duration for the validity of an open IOV, before to force a new query. 

#### PR validation:

Integration tests. The existing ones have been reviewed, and 2 novel tests have been added.

### CODE REVIEW

File: CondCore/CondDB/interface/IOVProxy.h [hunk 1]
```diff
@@ -122,8 +122,8 @@ namespace cond {
       // otherwise, a new query will be executed using the resolved group boundaries.
       // throws if the target time cannot be found.
       cond::Iov_t getInterval(cond::Time_t time);
-      cond::Iov_t getInterval(cond::Time_t time, cond::Time_t defaultIovSize);
 
+      //
```
Comment (tvami) [inline CondCore/CondDB/interface/IOVProxy.h:126]: was this meant to be a beginning of a comment?

Comment (ggovi) [reply CondCore/CondDB/interface/IOVProxy.h]: not sure :-) - removed now

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by yuanchao** [comment]
+1
* unit tests passed
* SA build warnings are not part of this PR modifications

**Comment by tvami** [comment]
Hi @ggovi I dont see the new unit tests here
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-860654/21413/unitTests/success.html

**Comment by ggovi** [comment]
@tvami None of the unit tests are currently running automatically for this package - it will require some more work, in a next PR...

**Comment by perrotta** [comment]
please test
(25 days old tests need to be refreshed...)

**Comment by perrotta** [comment]
+1

**Pull request merged**